relay-server/src/actors/processor.rs

```diff
@@ -3116,73 +3186,53 @@ mod tests {
         // None represents no TransactionMetricsConfig, DS will not be run
         let mut state = get_state(None);
         service.run_dynamic_sampling(&mut state);
-        assert!(matches!(state.sampling_result, SamplingResult::Keep));
+        assert!(state.sampling_result.should_keep());
 
         // Current version is 1, so it won't run DS if it's outdated
         let mut state = get_state(Some(0));
         service.run_dynamic_sampling(&mut state);
-        assert!(matches!(state.sampling_result, SamplingResult::Keep));
+        assert!(state.sampling_result.should_keep());
 
         // Dynamic sampling is run, as the transactionmetrics version is up to date.
         let mut state = get_state(Some(1));
         service.run_dynamic_sampling(&mut state);
-        assert!(matches!(state.sampling_result, SamplingResult::Drop(_)));
+        assert!(state.sampling_result.should_drop());
     }
 
-    #[tokio::test]
-    async fn test_it_keeps_or_drops_transactions() {
-        relay_test::setup();
-
-        let (outcome_aggregator, test_store) = services();
-
-        // an empty json still produces a valid config
-        let json_config = serde_json::json!({});
-
-        let config = Config::from_json_value(json_config).unwrap();
-
-        let service = create_test_processor(config);
-
+    #[test]
```

Comment:
changing the signature of `compute_sampling_decision` made this test a lot simpler